src/screens/transaction-details/components/TransactionDetailsValueAndFeeSection.tsx

```diff
@@ -26,18 +26,20 @@ export const TransactionDetailsValueAndFeeSection: React.FC<Props> = ({
   transaction,
 }) => {
   const { network, symbol, type, fee } = transaction;
-  // FIXME: Doesn't fully support Layer 2 Networks yet and not all coin icons work properly.
-  const coinSymbol =
-    type === TransactionType.contract_interaction
-      ? ethereumUtils.getNetworkNativeAsset(network ?? Network.mainnet)?.symbol
-      : symbol ?? undefined;
-  const mainnetCoinAddress = useSelector(
+  const assetData = useSelector(
     (state: AppState) =>
       state.data.accountAssetsData?.[
         `${transaction.address}_${transaction.network}`
-      ]?.mainnet_address
+      ]
   );
-  const coinAddress = mainnetCoinAddress ?? transaction.address ?? undefined;
+  const coinAddress = assetData?.address ?? transaction.address;
+  const mainnetCoinAddress = assetData?.mainnet_address;
+  const coinSymbol =
+    type === TransactionType.contract_interaction
+      ? ethereumUtils.getNetworkNativeAsset(network ?? Network.mainnet)?.symbol
+      : assetData?.symbol ?? symbol ?? undefined;
+  const coinType =
+    assetData?.type ?? network !== Network.mainnet ? network : undefined;
```

Comment:
I think the last fallback should be AssetType.token no?  which would be mainnet assets

Reply:
Yeah I think it can be like this, I am not sure if it is not already a fallback in coin icon component, but better to be explicit here I think